### ISSUE TITLE
Fails during headphones jack in / jack out

### ISSUE THREAD

**Issue by MikhailRokhin** [fxsound2/fxsound-app#201]
Control Panel\System and Security\Security and Maintenance\Problem Details :

Source
FxSound

Summary
Stopped working

Date
‎01.‎11.‎2024 23:04

Status
Report sent

Description
Faulting Application Path:	C:\Program Files\FxSound LLC\FxSound\FxSound.exe

Problem signature
Problem Event Name:	APPCRASH
Application Name:	FxSound.exe
Application Version:	1.1.27.0
Application Timestamp:	66c61bf9
Fault Module Name:	FxSound.exe
Fault Module Version:	1.1.27.0
Fault Module Timestamp:	66c61bf9
Exception Code:	c0000005
Exception Offset:	00000000000b4ec6
OS Version:	10.0.22631.2.0.0.768.101
Locale ID:	1049
Additional Information 1:	3500
Additional Information 2:	3500346866e1a839e5f2d5a2bb0c78ae
Additional Information 3:	60a5
Additional Information 4:	60a5740270796d6fdfdeb377b076b753

Extra information about the problem
Bucket ID:	097a10300d7ac7cbe73d34be1ff07c96 (1674552627635977366)

---

\
Also, suddenly, from time to time it loses volume and messes up it, setting speakers volume to headphones and vice versa, during headphones jack in / jack out.

**System information**
 - Windows version 10.0.22631.4317 
 - Audio device ``High Definition Audio Device ``
 - FxSound version 1.1.27.0 

Do you plan to recompile ``FxSound Audio Enhancer`` driver as it is recognized as ``Windows Vista`` by OS context?

Please, apply #AI automative extended stress-test of sample before release to all your software. 

Thank you for development & support.

Joy, Happiness and Satisfaction! Resources for prosperous life!!)

**Comment by raddyst** [comment]
Add crashdump (fxsound-*.dmp) from `%LOCALAPPDATA%\CrashDumps` to this report.

**Comment by MikhailRokhin** [comment]
> Add crashdump (fxsound-*.dmp) from `%LOCALAPPDATA%\CrashDumps` to this report.

[FxSound.exe.17600.dmp.zip](https://github.com/user-attachments/files/17606369/FxSound.exe.17600.dmp.zip)

**Comment by MikhailRokhin** [comment]
[FxSound.exe.10408.dmp.zip](https://github.com/user-attachments/files/17820165/FxSound.exe.10408.dmp.zip)

**Comment by michal-a11y** [comment]
hi @MikhailRokhin
## here`s some steps that help with the ploblem
 - try to reinstall fxsound, this may stop the ploblem from cause an appcrash if that ploblem still not fixed
 - install the latest or lastest stable verson of fxsound, this may work https://download.fxsound.com/fxsoundlatest
 -  if any of these steps wont work: ask one of Contributors for further support.

**Comment by MikhailRokhin** [comment]
The problem persists through versions up to `1.2.2.0` & needs manual switch via `Windows Settings\System\Sound`. Looks like it looses context & can't switch from Speakers to Headphones.

Also, suddenly, it can miss the volume switch, when jack out at hibernation and turn on notebook then.

**Comment by MikhailRokhin** [comment]
Version 1.2.5.0

[FxSound.exe.14900.dmp.zip](https://github.com/user-attachments/files/24651508/FxSound.exe.14900.dmp.zip)

**Comment by MikhailRokhin** [comment]
The problem persists through versions up to 1.2.6.0 & needs manual multiple reswitch jack in / jack out to achieve speakers-to-headphones.

**Comment by MikhailRokhin** [comment]
Exited without crashdump and error reports after wake-up from hibernation and jack in .

It can't switch at lock screen.

Upd: Found "C:\Users\user\AppData\Roaming\FxSound\fxsound.dmp" 

[fxsound.dmp](https://github.com/user-attachments/files/26514125/fxsound.dmp)

[fxsound.log](https://github.com/user-attachments/files/26514212/fxsound.log)

**Comment by bvijay74** [comment]
@MikhailRokhin The crash observed in v1.2.6.0 should have been fixed in the latest beta version v1.2.9.0.

**Comment by bvijay74** [comment]
Several device detections issues are fixed in v1.2.10.0

**Comment by MikhailRokhin** [comment]
> [@MikhailRokhin](https://github.com/MikhailRokhin) The crash observed in v1.2.6.0 should have been fixed in the latest beta version v1.2.9.0.

v1.2.9.0 does not switch from Speakers to Headphones via manual multiple reswitch jack in / jack out to achieve speakers-to-headphones

**Comment by bvijay74** [comment]
Device detection issues were fixed in v1.2.10.0.
Can you please check in v1.2.10.0?

**Comment by MikhailRokhin** [comment]
> Device detection issues were fixed in v1.2.10.0. Can you please check in v1.2.10.0?

No, I wait for v1.2.10.0 Release - Beta is too slow for response via user interface. 

Also it created 2 extra Speakers devices, ignoring existing one of 1.2.6.0 .

**Comment by MikhailRokhin** [comment]
> Device detection issues were fixed in v1.2.10.0. Can you please check in v1.2.10.0? 

The 1.2.10.0 release created extra Speakers device, ignoring existing one of 1.2.9.0 . There is no check for existing devices.

Manual uninstall & install from the scratch helped.

<img width="878" height="441" alt="Image" src="https://github.com/user-attachments/assets/f5e3334f-40f4-43e0-bedc-c59679f8fb2e" />

**Comment by MikhailRokhin** [comment]
Update from app failed (1.2.10.0 -> 1.2.11.0 )

<img width="571" height="513" alt="Image" src="https://github.com/user-attachments/assets/73d2c256-74a5-41ee-8721-35c7fb0e0f3d" />

**Comment by MikhailRokhin** [comment]
<img width="367" height="110" alt="Image" src="https://github.com/user-attachments/assets/159b6904-2d20-4872-b4a2-116a0f8a721f" />

<img width="767" height="470" alt="Image" src="https://github.com/user-attachments/assets/5eb9a845-e38c-4ca3-be5f-14087c36c3cc" />

Updated 1.2.11.0 -> 1.2.13.0
